src/main/java/org/jenkinsci/plugins/github/extension/GHEventsSubscriber.java

```diff
@@ -55,12 +55,13 @@
      * This method called when root action receives webhook from GH and this extension is interested in such
      * events (provided by {@link #events()} method). By default do nothing and can be overrided to implement any
      * parse logic
-     * Don't call it directly, use {@link #processEvent(GHEvent, String)} static function
+     * Don't call it directly, use {@link #processEvent(GHEvent, String, String)} static function
```

Comment:
@lanwen Considering that there are project-specific secrets, another solution would be to parse the payload twice as the validation would be dependent on the repo which is specified inside the payload, thus making it required to parse the message before validating. Do you think this sounds like an acceptable solution or do you have any other idea how to approach this?